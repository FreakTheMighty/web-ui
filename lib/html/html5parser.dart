// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.
// Generated by scripts/tree_gen.py.

/**
 * An HTML5 parser written in Dart.
 *
 * Intended to parse all html documents using the [html5 parsing algorithm][p],
 * although the implementation is not yet complete.
 *
 * [p]: http://dev.w3.org/html5/spec/parsing.html#parsing
 */
#library('html5parser');

#source('htmltree.dart');
#source('parser.dart');
#source('tree.dart');
#source('token.dart');
#source('tokenkind.dart');
#source('tokenizer_base.dart');
#source('tokenizer.dart');
