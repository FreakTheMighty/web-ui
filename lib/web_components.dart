// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

/**
 * **Deprecated**: import 'web_ui/web_ui.dart' instead.
 *
 * This library exposes the types in [watcher], [safe_html], [templating] and
 * the [WebComponent] base class. See this article for more information about
 * this library: <http://www.dartlang.org/articles/dart-web-components/>.
 */
@deprecated library web_components;

import 'package:meta/meta.dart';
export 'package:web_ui/web_ui.dart';
