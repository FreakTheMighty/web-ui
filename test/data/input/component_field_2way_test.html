<!doctype html>
<!--
   Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
   for details. All rights reserved. Use of this source code is governed by a
   BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script type="application/javascript" src="testing.js"></script>
  <style>x-greeter { display: block; }</style>
</head>
<body>
  <element name="x-greeter" extends="div" constructor="Greeter">
    <template></template>
    <script type='application/dart'>
      import 'package:web_components/web_components.dart';
      class Greeter extends WebComponent {
        String greeting;
        changeStuff() {
          greeting = 'Yo dawg';
          title = 'I heard you like $title so we put $title in yo $title.';
        }
      }
    </script>
  </element>
  <p>
  The message changes from "hello world" to something more interesting.
  </p>
  <!-- Note: "title" is a DOM field on all elements.
       Also this element has no rendering. -->
  <x-greeter id="greet" bind-title="data[1]" bind-greeting="data[0]">
  </x-greeter>
  <p id="actual">{{data[0]}} {{data[1]}}</p>
  <script type="application/dart">
    import 'dart:html';
    import 'package:unittest/unittest.dart';
    import 'package:web_components/web_components.dart';

    var data = ['hello', 'world'];
    main() {
      window.setTimeout(() {
        var actual = query('#actual');
        var greet = query('#greet');
        expect(actual.innerHTML, 'hello world');
        greet.xtag.changeStuff();
        dispatch();
        expect(actual.innerHTML, contains('Yo dawg I heard'));
        window.postMessage('done', '*');
      }, 0);
    }
  </script>
</body>
</html>
