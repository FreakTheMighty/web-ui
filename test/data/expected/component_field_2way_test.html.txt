Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x88
  RenderBlock {HTML} at (0,0) size 800x88
    RenderBody {BODY} at (8,16) size 784x56
      RenderBlock {P} at (0,0) size 784x20
        RenderText {#text} at (0,0) size 426x19
          text run at (0,0) width 426: "The message changes from \"hello world\" to something more interesting."
      RenderBlock {X-GREETER} at (0,36) size 784x0
      RenderBlock {P} at (0,36) size 784x20
        RenderText {#text} at (0,0) size 56x19
          text run at (0,0) width 56: "Yo dawg"
        RenderText {#text} at (56,0) size 4x19
          text run at (56,0) width 4: " "
        RenderText {#text} at (60,0) size 311x19
          text run at (60,0) width 311: "I heard you like world so we put world in yo world."
